### PR TITLE
Keep blocking until Cromwell query responds

### PR DESCRIPTION
**Pull request by geoffjentry**
Apparently the incoming Akka HTTP server spins up a little less quickly (or a little more quickly?) than the Spray stuff. Centaur had been relying on the version endpoint to detect if Cromwell was alive, but it would regularly lead to real requests being fired over during liquibasing and causing timeouts.

Instead, use the query endpoint to check for liveness. The query endpoint manages to hit the DB w/o requiring a workflow ID and is also idempotent.

### CODE REVIEW

**Comment by Horneth** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/220/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/scala/centaur/api/CentaurCromwellClient.scala [hunk 1]
```diff
@@ -36,7 +38,14 @@ object CentaurCromwellClient {
     sendReceiveFutureCompletion(() => cromwellClient.status(workflow.id))
   }
 
-  def isAlive: Boolean = Try(Await.result(cromwellClient.version, CentaurConfig.sendReceiveTimeout)).isSuccess
+  /*
+    Sends a quick ping to the Cromwell query endpoint. The query endpoint is the only one which both hits the
+    database w/o requiring a workflow id and is idempotent. Not using CromwellClient here as it currently does not support query.
```
Comment (mcovarr) [inline src/main/scala/centaur/api/CentaurCromwellClient.scala]: "idempotent" isn't wrong, but actually this endpoint is "safe" in that it doesn't change server state at all.

Comment (mcovarr) [reply src/main/scala/centaur/api/CentaurCromwellClient.scala]: sucks about query not being supported, I didn't know that  😢

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/220/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

**Pull request merged**